endpoint.go

```diff
@@ -303,14 +306,16 @@ func (ep *endpoint) Join(containerID string, options ...EndpointOption) (*Contai
 		}
 	}
 
-	err = sb.SetGateway(joinInfo.gw)
-	if err != nil {
-		return nil, err
-	}
+	if primary {
```

Comment:
This will be set only for the implicitly created endpoint and there can be only one implicitly created endpoint per container. And we don't allow it to be set via API or UI. This assumes that we are not going to support multiple `--net` options in `docker run`. That has other problems anyways so it's not a good idea anyways. With all this context, there can't be multiple endpoints with primary joining the same container. Even if it did it will fail right here when we try to set the default gateway.